src/pypi_attestations/_impl.py

```diff
@@ -388,6 +414,33 @@ class GitHubPublisher(_PublisherBase):
     action was performed from.
     """
 
+    def _as_policy(self) -> VerificationPolicy:
+        policies: list[VerificationPolicy] = [
+            policy.OIDCIssuerV2("https://token.actions.githubusercontent.com"),
+            policy.OIDCSourceRepositoryURI(f"https://github.com/{self.repository}"),
+        ]
+
+        if not self.claims:
+            raise VerificationError("refusing to build a policy without claims")
+
+        sha = self.claims.get("sha")
+        ref = self.claims.get("ref")
+        if not (sha or ref):
+            # This should never happen, since we should always _at least_ have
+            # the `sha` claim.
+            raise VerificationError("refusing to build a policy without a sha or ref claim")
```

Comment:
Will this work? As far as I can tell, `warehouse` is not including the claims in the publisher data inside the Provenances:
https://github.com/pypi/warehouse/blob/10570d141ceb7377b374b425f80a4bf284df126c/warehouse/attestations/services.py#L56-L67

Reply:
This is what I get when I download the provenance for `abi3info`:
```
Collecting abi3info
  Downloading abi3info-2024.10.8-py3-none-any.whl.metadata (3.6 kB)
kind='GitHub' claims=None repository='woodruffw/abi3info' workflow='release.yml' environment=''
```

Reply:
Hmm, I thought we were. That's not great, we should fix that.

In that case though, I need to do this the "hard" way and write a new `VerificationPolicy` that'll perform the appropriate partial match on the URL here.

Thanks for pointing this out!